alacritty_terminal/src/term/mod.rs

```diff
@@ -830,13 +920,122 @@ impl SizeInfo {
             col: min(col, Column(self.cols().saturating_sub(1))),
         }
     }
+
+    #[inline(always)]
+    pub fn into_u32(self) -> (u32, u32, u32, u32, u32, u32) {
+        (
+            self.width as u32,
+            self.height as u32,
+            self.cell_width as u32,
+            self.cell_height as u32,
+            self.padding_x as u32,
+            self.padding_y as u32,
+        )
+    }
 }
 
 impl<T> Term<T> {
     pub fn selection(&self) -> &Option<Selection> {
         &self.grid.selection
     }
 
+    fn calculate_selection_damage(
```

Comment:
Could we just have damage_selection and damage_hightlight methods as before, without this funny return type?

Reply:
It's a borrow-checker workaround. Couldn't satisfy the borrow checker with the old construct.

In the previous iteration, what was mutated was a clone of damage_list, not a member of Term. I could not pass a reference to a Term field to a mutable method on the same Term, due to the argument first creating an immutable borrow, which rules out having the method call make its mutable borrow.

I don't want to pointlessly clone and swap the damage struct on every render, but prettier workarounds welcome.

Reply:
You don't need to pass a vector. Just use `&mut self` and access it directly from a terminal.

Reply:
`self.damage_selection(&self.grid.selection)`or `self.damage_selection(&self.grid.selection, &mut self.damage)` does not work, as that tries to take an immutable borrow on a self field, and then create mutable borrows on the whole of self.

The previous iteration used `self.damage_selection(&self.grid.selection, &mut damage_list)`, where damage_list was swapped out of `self`.

The only alternative I can think of would be to experiment with having `damage_selection` iterate over all the fields by making an array of references containing them, but that might still trip the borrow checker. Regardless of what we do, it's not going to be as pretty as the previous iteration. :/

Reply:
nah, we can potentially use `RefCell` here, but lets just return Option<Range<Point>> for now.

Reply:
Well, it's not a Range, but rather a rect defined by two opposing corners, upper left and lower right. I changed it to Option<(Point, Point)>, which makes code using it much clearer.